### PR TITLE
feat: add python project type

### LINKED ISSUE #48: Support reading the current version from a python setup.cfg
### Feature Request

Support reading / writing the version from a setup.cfg from a python project

An example of this file could look like...

```
[metadata]
name = mypackage
version = 0.1.23
url = http://www.example.com/mypackage
author = Fake author
author_email = fake@example.com
description = My Package does something very interesting
long_description = file: ABOUT.rst
license = University of Illinois/NCSA Open Source License
license_file = LICENSE
classifiers =
    Programming Language :: Python :: 3.7
    License :: OSI Approved :: University of Illinois/NCSA Open Source License
    Topic :: System :: Archiving :: Packaging
```

### PR DESCRIPTION
**Pull request by ADI10HERO**
fixes #48 

This PR: 
1. Adds skeleton code for python project type
2. Tests for the same

Pending work (to be fixed/merged with next PR)
1. Adding support via `pyproject.toml` see #49 
2. Update current version reading from `setup.py` (discussion needed)

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/test/java/io/jenkins/plugins/conventionalcommits/utils/CurrentVersionTest.java [hunk 1]
```diff
@@ -155,6 +155,69 @@ public void testNpmProjectVersion() throws IOException, InterruptedException {
     assertThat(actualCurrentVersion, is(testCurrentVersion));
   }
 
+  @Test
+  public void testPythonProjectVersion_SetupPyExists() throws IOException, InterruptedException {
+
+    String os = System.getProperty("os.name");
+    String commandName = "python";
```
Comment (kwhetstone) [inline src/test/java/io/jenkins/plugins/conventionalcommits/utils/CurrentVersionTest.java:162]: Not something you need to have for this PR, but you could pull this out to a configuration so the command is the same for the test and for the main class.

**Review by kwhetstone — APPROVED** [review]
I think this looks good.  It's just important to document how to use the step and the files that the python step is looking for.

File: src/main/java/io/jenkins/plugins/conventionalcommits/utils/PythonProjectType.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.apache.commons.lang.NotImplementedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
+
+public class PythonProjectType extends ProjectType {
```
[the remark this answers is not in the document]
Comment (ADI10HERO) [reply src/main/java/io/jenkins/plugins/conventionalcommits/utils/PythonProjectType.java:14]: Yes, I will add this when I push the next PR related to python :)

File: src/test/java/io/jenkins/plugins/conventionalcommits/utils/CurrentVersionTest.java (hunk 1 above)
Comment (ADI10HERO) [reply src/test/java/io/jenkins/plugins/conventionalcommits/utils/CurrentVersionTest.java:162]: Do you mean having a `config.java` or something from where I can just import and use the command? (for this and other project types as well?)

**Pull request merged**